[torchbind] Better error message when missing init.

suo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#37474 [torchbind] Better error message when missing init.**
* #37473 [jit] move torchbind tests to separate file

Previously we would segfault

Differential Revision: [D21297542](https://our.internmc.facebook.com/intern/diff/D21297542)

suo committed:
Better error message when missing init.

Previously we would segfault

[ghstack-poisoned]

suo committed:
Update on "Better error message when missing init."


Previously we would segfault

[ghstack-poisoned]

suo committed:
Update on "[torchbind] Better error message when missing init."


Previously we would segfault

[ghstack-poisoned]

suo committed:
Update on "[torchbind] Better error message when missing init."


Previously we would segfault

Differential Revision: [D21297542](https://our.internmc.facebook.com/intern/diff/D21297542)

[ghstack-poisoned]

suo committed:
Update on "[torchbind] Better error message when missing init."


Previously we would segfault

Differential Revision: [D21297542](https://our.internmc.facebook.com/intern/diff/D21297542)

[ghstack-poisoned]

jamesr66a reviewed (approved):
LGTM. In TorchScript this just shows up as a compile error, right?

suo committed:
Update on "[torchbind] Better error message when missing init."


Previously we would segfault

Differential Revision: [D21297542](https://our.internmc.facebook.com/intern/diff/D21297542)

[ghstack-poisoned]

suo committed:
Update on "[torchbind] Better error message when missing init."


Previously we would segfault

Differential Revision: [D21297542](https://our.internmc.facebook.com/intern/diff/D21297542)

[ghstack-poisoned]

suo committed:
Update on "[torchbind] Better error message when missing init."


Previously we would segfault

Differential Revision: [D21297542](https://our.internmc.facebook.com/intern/diff/D21297542)

[ghstack-poisoned]

suo committed:
Update on "[torchbind] Better error message when missing init."


Previously we would segfault

Differential Revision: [D21297542](https://our.internmc.facebook.com/intern/diff/D21297542)

[ghstack-poisoned]

suo committed:
Update on "[torchbind] Better error message when missing init."


Previously we would segfault

Differential Revision: [D21297542](https://our.internmc.facebook.com/intern/diff/D21297542)

[ghstack-poisoned]

suo committed:
Update on "[torchbind] Better error message when missing init."


Previously we would segfault

Differential Revision: [D21297542](https://our.internmc.facebook.com/intern/diff/D21297542)

[ghstack-poisoned]

suo committed:
Update on "[torchbind] Better error message when missing init."


Previously we would segfault

Differential Revision: [D21297542](https://our.internmc.facebook.com/intern/diff/D21297542)

[ghstack-poisoned]

jeffdaily commented:
@suo Please verify ROCm CI prior to merging.  In this case, the newly added test needed to be skipped on ROCm (see other tests in the same file).  #38488 was needed to unblock CI.
